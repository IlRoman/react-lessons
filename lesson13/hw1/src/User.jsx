import React from 'react';

class User extends React.Component {
    state = {
        userName: '',
        userAvatar: '',
        userLocation: '',
    }

    componentDidUpdate(prevProps, prevState) {
        if (prevProps.match.url !== this.props.match.url) {
            this.fetchUserData(this.props.match.url);
        }
    }

    componentDidMount() {
        this.fetchUserData(this.props.match.url);
    }

    fetchUserData = (path) => {
        fetch(`https://api.github.com${path}`)
            .then(response => response.json())
            .then(user => this.setState({
                userName: user.name,
                userAvatar: user.avatar_url,
                userLocation: user.location
            }));
    }

    render() {
        const { userName, userLocation, userAvatar } = this.state;
        return (
            <div className="user">
                <img alt="User Avatar" src={userAvatar} className="user__avatar" />
                <div className="user__info">
                    <span className="user__name">{userName}</span>
                    <span className="user__location">{userLocation}</span>
                </div>
            </div>
        );
    }
};

export default User;