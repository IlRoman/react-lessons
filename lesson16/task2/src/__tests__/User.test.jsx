import React from 'react';
import User from '../User';
import { shallow } from 'enzyme';
import { getUserData } from '../usersGateway';

jest.mock('../usersGateway', () => {
    return {
        getUserData: jest.fn(() => Promise.resolve())
    }
});

describe('User', () => {
    it('should not display user', () => {
        const wrappedComponent = shallow(<User />);
        expect(wrappedComponent.find('.user').exists()).toEqual(false);
    });

    it("should display user's data", () => {
        const wrappedComponent = shallow(<User />);
        wrappedComponent.setState({
            userData: {
                avatar: 'https://examrle.com',
                name: 'John',
                location: 'Planet Earth'
            }
        });
        expect(wrappedComponent.find('.user')).toMatchSnapshot();
    });

    it("should display user's data", () => {
        shallow(<User />);
        expect(getUserData).toBeCalled();
    });
});