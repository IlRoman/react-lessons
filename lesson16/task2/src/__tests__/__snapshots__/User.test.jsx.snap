// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should display user's data 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      John
    </span>
    <span
      className="user__location"
    >
      Planet Earth
    </span>
  </div>
</div>
`;
